### DISCUSSION TITLE
Browser on RPi setting won't stay enabled

### DISCUSSION THREAD

**Discussion by ceejayemm** [rern/rAudio#1234] [Fixed Issues] [answered]
I have I installed the RPi 2 (32 bit) i20230212 image on my RP1 3B today which has a Raspberry Official 7in touchscreen attached.  Unfortunately although I can choose the settings for this in Features / Browser on RPi when I exit the setting dialogue screen for this the Browser on RPi switch is always returns to Disabled / Off.  This means I cannot use this screen with rAudio-1 right now.

Regards.

Chris

**Comment by ceejayemm** [comment]
Further information:

chromium 101.0.4951.41-1
# cat /srv/http/data/system/localbrowser.conf
rotate=NORMAL
zoom=70
screenoff=10
onwhileplay=false
cursor=false

localbrowser.service- Browser on RPi
     Loaded: loaded (/etc/systemd/system/localbrowser.service; disabled; preset: disabled)
     Active: inactive (dead)

Feb 12 16:29:58 RAConservatory sudo[3462]: modprobe: FATAL: Module g2d_23 not found in directory /lib/modules/6.1.11-1-rpi-ARCH
Feb 12 16:30:00 RAConservatory sudo[3482]: Failed to connect to bus: No medium found
Feb 12 16:30:00 RAConservatory sudo[3483]: dbus-update-activation-environment: error: unable to connect to D-Bus: Using X11 for dbus-daemon autolaunch was disabled at compile time, set your DBUS_SESSION_BUS_ADDRESS instead
Feb 12 16:30:00 RAConservatory sudo[3495]: /usr/lib/chromium/chromium: error while loading shared libraries: libicui18n.so.71: cannot open shared object file: No such file or directory
Feb 12 16:30:00 RAConservatory sudo[3460]: xinit: connection to X server lost
Feb 12 16:30:00 RAConservatory sudo[3460]: 
Feb 12 16:30:00 RAConservatory sudo[3460]: waiting for X server to shut down
Feb 12 16:30:00 RAConservatory sudo[3461]: (II) Server terminated successfully (0). Closing log file.
Feb 12 16:30:01 RAConservatory sudo[3442]: pam_unix(sudo:session): session closed for user root
Feb 12 16:30:01 RAConservatory systemd[1]: localbrowser.service: Deactivated successfully.

Seems like it might be an issue with running Chromium service itself rather that a settings problem.

Chris

**Comment by rern** [comment]
Hi Chris,

Chromium for 32bit has been stuck with version 101.x... since last April. The latest for 64bit is 110.x... which runs fine. I think it's about time to switch to Firefox.

Please try the UPDATE branch which will remove Chromium and install Firefox instead.

**Comment by ceejayemm** [comment]
@rern 

Thanks for this quick update.  I have applied the Update branch and can see that Chromium was removed and Firefox installed.  However when I try to enable the Browser on RPi option nothing happens and clicking on the 'i' info icon gives:

firefox 109.0.1-1
# cat /srv/http/data/system/localbrowser.conf
rotate=NORMAL
zoom=70
screenoff=10
onwhileplay=false
cursor=false

localbrowser.service- Browser on RPi
     Loaded: loaded (/etc/systemd/system/localbrowser.service; disabled; preset: disabled)
     Active: inactive (dead)

I have reboot the RPi and still the same.

Chris

**Reply by rern** [reply]
Hi Chris,

Please try again.

**Comment by ceejayemm** [comment, marked as answer]
@rern 

Thanks, the latest Update branch version does seem to fix the problems with using Firefox as the RPi browser in the 32 bit implementation.

Being cheeky I had hoped that the touchscreen issues identified in #426 might have also been fixed by the move to Firefox.  Unfortunately this is not the case.  It still seems that on the Settings screens (under the Cog icon) wherever a right hand scroll bar is present then it is not possible to exit from the Settings screen as the scrollbar seems to overlay, and take precedence over, the the 'X' Exit icon.  It also seems to overlay the '?' help icon too. As mentioned previously in the original discussion may be an alternative (or a duplicated) 'X and '?' icon screen location might help ?

Regards and thanks.

Chris
